app.py

```diff
@@ -1100,6 +1103,187 @@ def teacher_tutorial(user):
+
+    # Add the available levels to the customizations dict -> simplify
+    # implementation on the front-end
+    customizations['available_levels'] = available_levels
+    cheatsheet = COMMANDS[g.lang].get_commands_for_level(level, g.keyword_lang)
+
+    load_customized_adventures(level, customizations, adventures)
+    load_saved_programs(level, adventures, loaded_program)
+    initial_tab = adventures[0].short_name
+
+    if loaded_program:
+        # Make sure that there is an adventure(/tab) for a loaded program, otherwise make one
+        initial_tab = loaded_program.adventure_name
+        if not any(a.short_name == loaded_program.adventure_name for a in adventures):
+            adventures.append(Adventure(
+                short_name=loaded_program.adventure_name,
+                # This is not great but we got nothing better :)
+                name=gettext('your_program'),
+                text='',
+                example_code='',
+                start_code=loaded_program.code,
+                save_name=loaded_program.name,
+                is_teacher_adventure=False,
+                is_command_adventure=loaded_program.adventure_name in KEYWORDS_ADVENTURES
+            ))
+
+    adventures_map = {a.short_name: a for a in adventures}
+
+    enforce_developers_mode = False
+    if 'other_settings' in customizations and 'developers_mode' in customizations['other_settings']:
+        enforce_developers_mode = True
+
+    hide_cheatsheet = False
+    if 'other_settings' in customizations and 'hide_cheatsheet' in customizations['other_settings']:
+        hide_cheatsheet = True
+
+    parsons = True if PARSONS[g.lang].get_parsons_data_for_level(level) else False
+    quiz = True if QUIZZES[g.lang].get_quiz_data_for_level(level) else False
+    tutorial = True if TUTORIALS[g.lang].get_tutorial_for_level(level) else False
+
+    quiz_questions = 0
+    parson_exercises = 0
+
+    if quiz:
+        quiz_questions = len(QUIZZES[g.lang].get_quiz_data_for_level(level))
+    if parsons:
+        parson_exercises = len(PARSONS[g.lang].get_parsons_data_for_level(level))
+
+    if 'other_settings' in customizations and 'hide_parsons' in customizations['other_settings']:
+        parsons = False
+    if 'other_settings' in customizations and 'hide_quiz' in customizations['other_settings']:
+        quiz = False
+
+    # max_level = hedy.HEDY_MAX_LEVEL
+    level_number = int(level)
+
+    commands = hedy.commands_per_level.get(level)
+    return render_template(
+        "code-page.html",
+        level_nr=str(level_number),
+        level=level_number,
+        current_page='Hour of Code',
+        prev_level=None,
+        next_level=None,
+        # prev_level=level_number - 1 if level_number > 1 else None,
+        # next_level=level_number + 1 if level_number < max_level else None,
+        # customizations=customizations,
+        hide_cheatsheet=hide_cheatsheet,
+        # enforce_developers_mode=enforce_developers_mode,
```

Comment:
This way of disabling parameters by commenting them out is not what we commonly do (I think setting the variable to False occurs more often) I actually think what you do here is more clear, you have the information that it is not in use right in view, just want to point out that this is not what is used elsewhere (maybe it should be?)

Reply:
Overall in the codebase indeed that's not the case. For all PRs i worked on, I had to find out where the variables are passed to the view and where not, that made it harder to find why certain part of the view renders or doesn't. 

I think we should comment them out to inform other devs that these may be used, see https://jinja.palletsprojects.com/en/3.0.x/templates/#variables
> If a variable or attribute does not exist, you will get back an undefined value. What you can do with that kind of value depends on the application configuration: the default behavior is to evaluate to an empty string if printed or iterated over, and to fail for every other operation.


So, if we assign an unused variable to false and we iterate over it, the rendering will fail. Whereas it wouldn't if we didn't assign it at all.

Reply:
Great find, thanks! Shall we open an issue to make other renderings similar (not even for you, this is a great first issue for a newcomer too)